code/modules/reagents/reagent_containers/dropper.dm

```diff
@@ -9,13 +9,21 @@
 	amount_per_transfer_from_this = 5
 	possible_transfer_amounts = list(1,2,3,4,5)
 	volume = 5
-	var/filled = 0
+
+/obj/item/weapon/reagent_containers/dropper/on_reagent_change()
+	update_icon()
+
+/obj/item/weapon/reagent_containers/dropper/update_icon()
+	if(reagents.total_volume<=0)
+		icon_state = "[initial(icon_state)]"
+	else
+		icon_state = "[initial(icon_state)]1"

```

Comment:
Why not just put all of this under `on_reagent_change`? Instead of having a proc that calls another proc? 

This is just unnecessary proc call overhead...yeah, it's not going to really impact anything, but is there an explicit reason this needs to be its own proc?